lib/derivations.nix

```diff
@@ -98,4 +98,7 @@ in
       # `lazyDerivation` caller knew a shortcut, be taken from there.
       meta = args.meta or checked.meta;
     } // passthru;
+
+  inspectMkDerivationArgs = drv:
+    (drv.overrideAttrs (o: { passthru.__inspectArgs = o; })).__inspectArgs;
```

Comment:
Not sure if relevant for this PR, but  sometimes it is useful to extract arguments of other layers of package functions. Therefore a more generic implementation could be useful here:
```nix
inspectDerivationArgsVia = funcName: drv:
    (drv.${funcName} (o: { passthru.__inspectArgs = o; })).__inspectArgs;
    
inspectMkDerivationArgs = inspectDerivationArgsVia "overrideAttrs";
```

Reply:
Not needed anymore, since we have `internals`. I don't think the extracted function would generalize particularly well, and I think we're trying to move away from layers of functions anyway. It's an interesting idea though and something like this could be helpful for `pkg.internals` as well, but we'll be free to change those values as we please anyway. That's kind of the point :)